### PR TITLE
policy: Use ClusterInfo in outbound indexer

### PR DESCRIPTION
**Pull request by olix0r**
We have a `ClusterInfo` type that includes all cluster-level configuration. The outbound indexer does not use this type and, instead, passes around individual configuration values.

This change updates the outbound indexer to use the ClusterInfo configuration type. It also moves the type to its own module.

This is largely a reorganization PR, though there is one notable change:
InvalidDst response no longer reference a backend. Instead, the backend
is left empty, since it's not possible to route requests to it.

### CODE REVIEW

File: policy-controller/k8s/index/src/cluster_info.rs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            "{}.{}.serviceaccount.identity.{}.{}",
+            sa, ns, self.control_plane_ns, self.identity_domain
+        )
+    }
+
+    pub(crate) fn namespace_identity(&self, ns: &str) -> String {
+        format!(
+            "*.{}.serviceaccount.identity.{}.{}",
+            ns, self.control_plane_ns, self.identity_domain
+        )
```
Comment (kleimkuhler) [inline policy-controller/k8s/index/src/cluster_info.rs:49]: I know this is just moving code around but we could use the other function here. Certainly not a blocker.

```suggestion
        self.service_account_identity(ns, "*")
```

**Review by kleimkuhler — APPROVED** [review]
Overall looks good!

**Review by hawkw — APPROVED** [review]

**Review by adleong — APPROVED** [review]
Looks good, though I'm sad to lose the name of the backend in the error response.  Rather than a dst or authority, could the InvalidDst carry a more human readable service name or error message?  "backend is invalid: Service foo does not exist"

File: policy-controller/k8s/index/src/outbound_index.rs [hunk 2]
```diff
@@ -322,27 +328,71 @@ fn convert_backend(
...
 ) -> Option<Backend> {
     backend.backend_ref.map(|backend| {
-        if let Some(port) = backend
+        if !is_backend_service(&backend.inner) {
+            return Backend::InvalidDst {
+                weight: backend.weight.unwrap_or(1).into(),
+                message: format!(
+                    "unsupported backend type {group} {kind}",
+                    group = backend.inner.group.as_deref().unwrap_or("core"),
+                    kind = backend.inner.kind.as_deref().unwrap_or("core"),
```
Comment (adleong) [inline policy-controller/k8s/index/src/outbound_index.rs:337]: default should be "Service" I think

Comment (olix0r) [reply policy-controller/k8s/index/src/outbound_index.rs]: good catch -- my defaulting was off in the matcher below.

In this case, i've changed the value to `<empty>` -- in this case we know it's not a normal core Service; so we really don't know what the inferred kind should be if the group is specified.

**Pull request merged**